include/parallel/parallel_eigen.h

```diff
@@ -0,0 +1,179 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2023 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+#ifndef LIBMESH_PARALLEL_EIGEN_H
+#define LIBMESH_PARALLEL_EIGEN_H
+
+// libMesh includes
+#include "libmesh/libmesh_config.h"
+#include "libmesh/libmesh_common.h"
+#include "libmesh/int_range.h"
+
+// TIMPI includes
+#include "timpi/packing.h"
+
+// libEigen includes
```

Comment:
Pretty much everything from here down ought to be wrapped in `#ifdef LIBMESH_HAVE_EIGEN`